CSS Font Loading API: Remove FontFaceSet constructor

vitorroriz opened the pull request:
#### c58996cc188d1ca5f11283c6c6d078ec7d4583fa
<pre>
CSS Font Loading API: Remove FontFaceSet constructor
<a href="https://bugs.webkit.org/show_bug.cgi?id=276794">https://bugs.webkit.org/show_bug.cgi?id=276794</a>
<a href="https://rdar.apple.com/132031306">rdar://132031306</a>

Reviewed by Simon Fraser.

The FontFaceSet constructor was considered not useful
and is to be removed from the FontFaceSet interface
as per [1] resolution.

[1] <a href="https://github.com/w3c/csswg-drafts/issues/10390">https://github.com/w3c/csswg-drafts/issues/10390</a>

* LayoutTests/fast/text/font-face-api-preinstalled.html:
* LayoutTests/fast/text/font-face-set-document-expected.txt:
* LayoutTests/fast/text/font-face-set-document.html:
* LayoutTests/fast/text/font-face-set-javascript.html:
* LayoutTests/fast/text/font-loading-multiple-sets.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-font-loading/fontfaceset-add-css-connected-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-font-loading/fontfaceset-add-css-connected.html: Removed.
  Remove fontfaceset-add-css-connected.html as it fundamentally requires the FontFaceSet constructor to function. The test verifies that CSS-connected FontFace objects cannot be added to a separate FontFaceSet instance, but without the constructor, there is no way to create such a separate instance in a document context. This test scenario becomes impossible once the constructor is removed per the CSSWG resolution mentioned above [1].

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/css/FontFaceSet.idl:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultFontFaceSetConstructorEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultFontFaceSetConstructorEnabled):

Canonical link: <a href="https://commits.webkit.org/304912@main">https://commits.webkit.org/304912@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/56224a2ed2878da94bdb6504dc938d08028cdae7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136864 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9224 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48151 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144600 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89837 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b5c78371-2a70-4ee6-87ae-98d304b5a0c4) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138736 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9927 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9070 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104654 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0fcc8d39-95f9-48ca-a877-69ddedfef45e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139809 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7259 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122626 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85492 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e617b4a9-0301-4078-881d-0d228bf31abe) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/4594 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5188 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/128825 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116226 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40809 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147357 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/135350 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8907 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41379 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/113012 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8925 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7482 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113341 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28791 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6823 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118909 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63096 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8955 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36963 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/168130 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8676 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/72521 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43871 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8895 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8747 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3154,6 +3154,21 @@ FlexFormattingContextIntegrationEnabled:
+  humanReadableDescription: "Legacy FontFaceSet Constructor API"
+  defaultValue:
+    WebKitLegacy:
+      "PLATFORM(IOS_FAMILY)": WebKit::defaultFontFaceSetConstructorEnabled()
```
Why only on iOS? Don't we care about macOS apps too?

vitorroriz replied:
I've adopted the same approach as when [we removed `showModalDialog()`](https://commits.webkit.org/252759@main) 

I thought WebKit legacy MacOS apps wouldn't be an issue (WebKit still uses defaultFontFaceSetConstructorEnabled directly). If you think it is more reasonable I can also use `defaultFontFaceSetConstructorEnabled` here.

smfr replied:
I think removing the platform guard is better.

vitorroriz replied:
alright, fixed, thank you

vitorroriz committed:
CSS Font Loading API: Remove FontFaceSet constructor
https://bugs.webkit.org/show_bug.cgi?id=276794
rdar://132031306

Reviewed by Simon Fraser.

The FontFaceSet constructor was considered not useful
and is to be removed from the FontFaceSet interface
as per [1] resolution.

[1] https://github.com/w3c/csswg-drafts/issues/10390

* LayoutTests/fast/text/font-face-api-preinstalled.html:
* LayoutTests/fast/text/font-face-set-document-expected.txt:
* LayoutTests/fast/text/font-face-set-document.html:
* LayoutTests/fast/text/font-face-set-javascript.html:
* LayoutTests/fast/text/font-loading-multiple-sets.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-font-loading/fontfaceset-add-css-connected-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-font-loading/fontfaceset-add-css-connected.html: Removed.
  Remove fontfaceset-add-css-connected.html as it fundamentally requires the FontFaceSet constructor to function. The test verifies that CSS-connected FontFace objects cannot be added to a separate FontFaceSet instance, but without the constructor, there is no way to create such a separate instance in a document context. This test scenario becomes impossible once the constructor is removed per the CSSWG resolution mentioned above [1].

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/css/FontFaceSet.idl:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultFontFaceSetConstructorEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultFontFaceSetConstructorEnabled):

Canonical link: https://commits.webkit.org/304912@main
